x/mongo/driver/operation.go

```diff
@@ -198,6 +198,8 @@ type Operation struct {
 
 	// ServerAPI specifies options used to configure the API version sent to the server.
 	ServerAPI *ServerAPIOptions
+
+	cmdName string
```

Comment:
If we want to `UpdateSnapshotTime` _only_ when the command was `find`, `aggregate` or `distinct`, `readWireMessage` needs access to the initial command name. I added this private `cmdName` field to `Operation`, but this does not seem ideal to me. I could also pass the `cmdName` as a parameter to `roundTrip` and `readWireMessage`, but that doesn't seem much better IMO. Let me know what you think.

Reply:
The only slight disadvantage is that `cmdName` will remain unset on the operation until after `writeWireMessage`, and it only applies to OP_MSG. So it seems slightly preferable to go the route of adding `cmdName` parameter as an argument to `roundTrip` and `readWireMessage`. Though if that is an involved change it does not seem worth it. Commenting in the operation struct to note that `cmdName` is only set when serializing OP_MSG seems preferable.

@divjotarora do you have a strong opinion here?

Reply:
Just added a comment on the `cmdName` field for now.

Reply:
An alternate approach that doesn't require adding `cmdName` to `Operation` is to call `UpdateSnapshotTime` in the `ProcessResponseFn` for the `Find`, `Distinct`, and `Aggregate` operations.

E.g. update in `find.go`:
```go
func (f *Find) processResponse(info driver.ResponseInfo) error {
	f.session.UpdateSnapshotTime(info.ServerResponse) // Call UpdateSnapshotTime on the Find session.

	var err error
	f.result, err = driver.NewCursorResponse(info)
	return err
}
```

That approach wouldn't handle the case where a user is forming `Operation` calls manually and bypassing the `Find` type. Is that a case we care about supporting?

Reply:
Discussed offline: a great idea but we do want to support the case where a user creates their own "find", "aggregate" or "read" `Operation` or uses `db.RunCommand` to run a read operation in a snapshot-enabled session. So, only adding `UpdateSnapshotTime` to `ProcessResponseFn`s for `Find`, `Distinct`, and `Aggregate` will not be enough.